gateways/custom/storagegrid/start.go

```diff
@@ -67,28 +70,28 @@ type routeConfig struct {
 	sgConfig            *storageGrid
 	eventSource         *gateways.EventSource
 	eventSourceExecutor *StorageGridEventSourceExecutor
-	dataCh              chan []byte
-	doneCh              chan struct{}
-	errCh               chan error
 	startCh             chan struct{}
 }
 
+type endpoint struct {
```

Comment:
it seems a lot of this refactor touches every single gateway implementation and this struct seems to be pretty generic enough to exist in each separate implementation. We need to think about making our lives easier and isolating common functionality while potentially moving all the different implementations out of tree. We can take some lessons from the CNI and CSI projects in K8s space... the goal being we have an interface that all implementations must implement in order to communicate. We need to take opinionated positions at some point so we don't get lost updating everything on every PR

Reply:
Agreed. I'll create an issue for this and we can discuss the options to handle this.